### ISSUE TITLE
Error Code Standardization

### ISSUE THREAD

**Issue by pascallapointe** [spacebudz/lucid#168]
### Error Improvement proposal

Lucid currently throw message error, which is not a problem in itself, but I think that using an object of the following format would be better:

```typescript
{
    code: number; // 1
    error: string; // INSUFFICIENT_FUNDS
    message: string; // 'Insufficient input in transaction'
}
````

I think that using a error code and name would be better since it gives you the room to later change the error message if needed.

Using `if (e === 'Insufficient input in transaction')` currently does the trick, but it doesn't give me as much confidence that the string won't change over time as a well defined error structure would.

Going forward with this would be a breaking change, but there might be a way to use the toString() prototype on each error class to keep backward compatibility for current implementation.

**Comment by alessandrokonrad** [comment]
Yes this is a good idea and I thought about this before. Need to check if the errors thrown in the WebAssembly modules can be returned as json.
